### DISCUSSION TITLE
Black theme

### DISCUSSION THREAD

**Discussion by Diaskhan** [sumatrapdfreader/sumatrapdf#2555] [Ideas]
Hi people,
I have Idea, What about Dark theme ??
Ist possible for PDF text document, but not possible for Image OCR documents.

**Comment by kjk** [comment]
That would be possible with themes which is tracked in https://github.com/sumatrapdfreader/sumatrapdf/issues/898

It's not done because it's a lot of work to get all the details right (because theming Windows controls is hard) and not the highest priority.

**Reply by rafalhogent** [reply]
> That would be possible with themes which is tracked in #898
> 
> It's not done because it's a lot of work to get all the details right (because theming Windows controls is hard) and not the highest priority.

What about making 2 separate compilations? 
One with fixed light theme controls, other with fixed dark controls (tabs, menu's, bars, etc...?)

**Reply by kjk** [reply]
The hard part is making sure that all windows / dialogs and controls in the application reflect the theme. Switching between them is trivial so making separate build doesn't help to solve the hard problem.

**Comment by ermanueh02** [comment]
I think that it may not be the greatest priority, but I also think that it would be the very last feature to make this app even more perfect than it already is. 

I am a college student and I need this app to visualize my pdfs. As a student i've got to read many notes, and books and stuff (and comics sometimes xdddd). When it comes to speed, SumatraPDF is just the best thing there is. Not only that, but the compatibility with many kinds of archives is just outstanding mixed with the intrinsec simplicity of the app (without the slowing features and interfaces that readers like adobe include). 

That makes the app perfect, but there's one thing that annoys me (and even makes my eyes hurt). And that's just sitting in front of my desk at night, writing a LaTeX document, looking for some physical principle in wikipedia in perfect dark mode (using auto dark mode and chrome's dark reader) and then when I've got to open this book that I must read, or the exercises document, or some article my colleagues have sent me to check... BANG! all the pixels in my screen assemble and use all their power to produce the whitest of lights when my pupils had already got used to dark light.

I know that I can invert my docs in advanced options, but having to access that file to toggle dark or light mode everytime I want to switch between modes is just so annoying. Please, developers, all I'm asking is for you to add a button at the top of the screen to toggle dark or light mode or, even better, add the option to make it use windows' mode like many apps on the market.

I hope that in time, you'll update this app and include this feature amongst all the bug fixes and improvements, because I assure you that it'll be a gamechanger.

**Reply by GitHubRulesOK** [reply]
> access that file to toggle dark or light mode

just press `I` button for eye relief, however it's better to **enhance night mode** by reduce colour contrast so set background to muted grey not white and colours retain a lower gamut reducing eye strain. Its easy to use two start commands one for "bright as dawn" and one for "twilight zone".

**Reply by rafalhogent** [reply]
> I think that it may not be the greatest priority, but I also think that it would be the very last feature to make this app even more perfect than it already is.
> 
> I am a college student and I need this app to visualize my pdfs. As a student i've got to read many notes, and books and stuff (and comics sometimes xdddd). When it comes to speed, SumatraPDF is just the best thing there is. Not only that, but the compatibility with many kinds of archives is just outstanding mixed with the intrinsec simplicity of the app (without the slowing features and interfaces that readers like adobe include).
> 
> That makes the app perfect, but there's one thing that annoys me (and even makes my eyes hurt). And that's just sitting in front of my desk at night, writing a LaTeX document, looking for some physical principle in wikipedia in perfect dark mode (using auto dark mode and chrome's dark reader) and then when I've got to open this book that I must read, or the exercises document, or some article my colleagues have sent me to check... BANG! all the pixels in my screen assemble and use all their power to produce the whitest of lights when my pupils had already got used to dark light.
> 
> I know that I can invert my docs in advanced options, but having to access that file to toggle dark or light mode everytime I want to switch between modes is just so annoying. Please, developers, all I'm asking is for you to add a button at the top of the screen to toggle dark or light mode or, even better, add the option to make it use windows' mode like many apps on the market.
> 
> I hope that in time, you'll update this app and include this feature amongst all the bug fixes and improvements, because I assure you that it'll be a gamechanger.

> I think that it may not be the greatest priority, but I also think that it would be the very last feature to make this app even more perfect than it already is.
> 
> I am a college student and I need this app to visualize my pdfs. As a student i've got to read many notes, and books and stuff (and comics sometimes xdddd). When it comes to speed, SumatraPDF is just the best thing there is. Not only that, but the compatibility with many kinds of archives is just outstanding mixed with the intrinsec simplicity of the app (without the slowing features and interfaces that readers like adobe include).
> 
> That makes the app perfect, but there's one thing that annoys me (and even makes my eyes hurt). And that's just sitting in front of my desk at night, writing a LaTeX document, looking for some physical principle in wikipedia in perfect dark mode (using auto dark mode and chrome's dark reader) and then when I've got to open this book that I must read, or the exercises document, or some article my colleagues have sent me to check... BANG! all the pixels in my screen assemble and use all their power to produce the whitest of lights when my pupils had already got used to dark light.
> 
> I know that I can invert my docs in advanced options, but having to access that file to toggle dark or light mode everytime I want to switch between modes is just so annoying. Please, developers, all I'm asking is for you to add a button at the top of the screen to toggle dark or light mode or, even better, add the option to make it use windows' mode like many apps on the market.
> 
> I hope that in time, you'll update this app and include this feature amongst all the bug fixes and improvements, because I assure you that it'll be a gamechanger.

 
 You can try a SecureUX theme patcher for win10/11
 https://github.com/namazso/SecureUxTheme
 
 With for example this theme - rectify11:
 https://discord.com/channels/972654790536482896/972654792352616520
 
 this will probably reduce your pain in many places.
 
 Than you have to edit these lines in advanced options:
 FixedPageUI [ TextColor = #ffffff BackgroundColor = #212121 ]
 
 If you want to quickly switch between sumatra themes you can use my very simple project:
 https://github.com/rafalhogent/ThemeSwitcher-for-SumatraPDF
 
 Only one thing in sumatra which remains white are tabs, hopely Sumatra Team will soon allow users to change tab colors or let the tabs to follow the color-scheme from SumatraPDF-settings.txt file
 
 
![sumatra-in-dark](https://user-images.githubusercontent.com/92117516/203633027-c69f1bec-7f18-497e-a885-dd665ab5496f.jpg)

**Comment by ParboiL1971** [comment]
We need a dark theme👍
